backend/src/main/java/ai/verta/modeldb/project/FutureProjectServiceImpl.java

```diff
@@ -63,27 +66,80 @@ public void updateProjectDescription(
       StreamObserver<AddProjectAttributes.Response> responseObserver) {
-    super.addProjectAttributes(request, responseObserver);
+    try {
+      final var response =
+          futureProjectDAO
+              .logAttributes(
+                  LogAttributes.newBuilder()
+                      .setId(request.getId())
+                      .addAllAttributes(request.getAttributesList())
+                      .build())
+              .thenCompose(
+                  unused -> futureProjectDAO.getPreAccessProjectById(request.getId()), executor)
+              .thenApply(
+                  project -> AddProjectAttributes.Response.newBuilder().setProject(project).build(),
+                  executor);
+      FutureGrpc.ServerResponse(responseObserver, response, executor);
+    } catch (Exception e) {
+      CommonUtils.observeError(responseObserver, e);
+    }
   }
 
   @Override
   public void updateProjectAttributes(
       UpdateProjectAttributes request,
       StreamObserver<UpdateProjectAttributes.Response> responseObserver) {
-    super.updateProjectAttributes(request, responseObserver);
+    try {
+      final var response =
+          futureProjectDAO
+              .updateProjectAttributes(request)
+              .thenCompose(
+                  unused -> futureProjectDAO.getPreAccessProjectById(request.getId()), executor)
+              .thenApply(
+                  project ->
+                      UpdateProjectAttributes.Response.newBuilder().setProject(project).build(),
+                  executor);
+      FutureGrpc.ServerResponse(responseObserver, response, executor);
+    } catch (Exception e) {
+      CommonUtils.observeError(responseObserver, e);
+    }
   }
 
   @Override
   public void getProjectAttributes(
       GetAttributes request, StreamObserver<GetAttributes.Response> responseObserver) {
-    super.getProjectAttributes(request, responseObserver);
+    try {
+      final var response =
+          futureProjectDAO
+              .getAttributes(request)
+              .thenApply(
+                  attributes ->
+                      GetAttributes.Response.newBuilder().addAllAttributes(attributes).build(),
+                  executor);
+      FutureGrpc.ServerResponse(responseObserver, response, executor);
+    } catch (Exception e) {
+      CommonUtils.observeError(responseObserver, e);
+    }
   }
 
   @Override
   public void deleteProjectAttributes(
       DeleteProjectAttributes request,
       StreamObserver<DeleteProjectAttributes.Response> responseObserver) {
-    super.deleteProjectAttributes(request, responseObserver);
+    try {
+      final var response =
+          futureProjectDAO
+              .deleteAttributes(request)
+              .thenCompose(
+                  unused -> futureProjectDAO.getPreAccessProjectById(request.getId()), executor)
+              .thenApply(
+                  project ->
+                      DeleteProjectAttributes.Response.newBuilder().setProject(project).build(),
```

Comment:
@hmacdonald-verta could you check with @convoliution that all of these APIs returning `Project` actually need them to return `Project`? I think this data is just discarded by the caller, so we could save some round trips here.

Reply:
The only time the client uses the `Project` returned in the response is in

- `/getProjectById`
- `/getProjectByName`
- `/createProject`

Erm...in fact, the python client doesn't even expose mutating attributes on projects, anyway.

Reply:
It doesn't look like the Scala client uses the returned `Project` for these endpoints, either.

Reply:
@conradoverta then should we ignore to feel project in the response??

Reply:
Yes. Let's also put up a PR to remove that from the return completely for clarity too.

Reply:
removed code for now in PR: https://github.com/VertaAI/modeldb/pull/2129